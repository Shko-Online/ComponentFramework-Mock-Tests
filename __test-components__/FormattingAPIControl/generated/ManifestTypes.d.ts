/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    currencyInput: ComponentFramework.PropertyTypes.NumberProperty;
    dateInput: ComponentFramework.PropertyTypes.DateTimeProperty;
    decimalInput: ComponentFramework.PropertyTypes.DecimalNumberProperty;
    integerInput: ComponentFramework.PropertyTypes.WholeNumberProperty;
}
export interface IOutputs {
    currencyInput?: number;
    dateInput?: Date;
    decimalInput?: number;
    integerInput?: number;
}
