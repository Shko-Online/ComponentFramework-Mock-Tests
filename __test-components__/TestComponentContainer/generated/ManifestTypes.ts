/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    controlValue: ComponentFramework.PropertyTypes.StringProperty;
}
export interface IOutputs {
    controlValue?: string;
}
