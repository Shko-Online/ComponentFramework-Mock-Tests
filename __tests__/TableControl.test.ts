/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

jest.useFakeTimers({
	now: new Date(2022, 9, 9, 10)
});

import * as sinon from 'sinon';

import { ComponentFrameworkMockGenerator, StringPropertyMock } from '@shko.online/componentframework-mock';
import { TableControl } from '@powerapps-samples/table-control/TableControl';
import { IInputs, IOutputs } from '@powerapps-samples/table-control/TableControl/generated/ManifestTypes';
import resource from '@powerapps-samples/table-control/TableControl/strings/TableControl.1033.resx';

describe("TableControl", () => {
	let mockGenerator: ComponentFrameworkMockGenerator<IInputs, IOutputs>;
	beforeEach(() => {
		const container = document.createElement('div');

		mockGenerator = new ComponentFrameworkMockGenerator(
			TableControl,
			{
				stringProperty: StringPropertyMock
			},
			container);
		mockGenerator.SetControlResource(resource);
		document.body.appendChild(container);
		mockGenerator.metadata.initMetadata([{
			LogicalName: 'account',
			Attributes: []
		} as ShkoOnline.EntityMetadata]);

		mockGenerator.context.utils.lookupObjects.callsFake((lookupOptions: ComponentFramework.UtilityApi.LookupOptions) => {
			return new Promise<ComponentFramework.LookupValue[]>((resolve) => {
				resolve([{
					entityType: lookupOptions.entityTypes ? lookupOptions.entityTypes[0] : 'account',
					id: "00000000-0000-0000-0000-000000000001",
					name: "Account",
				}])
			});
		})

	})

	afterEach(() => {
		document.body.innerHTML = null;
	})
	it("Init should work", () => {
		mockGenerator.ExecuteInit();
		sinon.assert.calledOnce(mockGenerator.control.init);
		expect(document.body).toMatchSnapshot();
	})
	it("Update View should Work", () => {
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		sinon.assert.calledOnce(mockGenerator.control.init);
		sinon.assert.calledOnce(mockGenerator.control.updateView);
		expect(document.body).toMatchSnapshot();
	})
	it("GetOutputs should work", () => {
		mockGenerator.control.getOutputs();
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot();
	})
	it("Destroy should work", () => {
		mockGenerator.control.destroy();
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot();
	})

	it("Click should Work", () => {

		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot();

		const buttons = mockGenerator.container.querySelectorAll("button");
		buttons.forEach((button) => {
			var evt = document.createEvent("Event");
			evt.initEvent("click", true, false);
			button.dispatchEvent(evt);

			mockGenerator.ExecuteUpdateView();
			expect(document.body).toMatchSnapshot();
		})

	})

	it("onlookup", () => {

		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		expect(document.body).toMatchSnapshot();
		const entitymeta = mockGenerator.context.utils.getEntityMetadata("account");
		const times = 0;
		const buttons = mockGenerator.container.querySelectorAll(".SampleControlHtmlTable_ButtonClass");
		buttons.forEach((button) => {
			var evt = document.createEvent("Event");
			evt.initEvent("click", true, false);
			button.dispatchEvent(evt);
			times + 1;

			mockGenerator.ExecuteUpdateView();
			expect(document.body).toMatchSnapshot();
		})
	})
})