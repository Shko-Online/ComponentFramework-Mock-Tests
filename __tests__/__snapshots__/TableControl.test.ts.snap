// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableControl Click should Work 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(true)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl Click should Work 2`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(false)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl Destroy should work 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(true)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl GetOutputs should work 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(true)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl Init should work 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    />
  </div>
</body>
`;

exports[`TableControl Update View should Work 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(true)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl onlookup 1`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(true)
      </button>
    </div>
  </div>
</body>
`;

exports[`TableControl onlookup 2`] = `
<body>
  <div>
    <div
      class="Table_Container"
    >
      <table
        class="SampleControlHtmlTable_HtmlTable"
      >
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Key"
          >
            Example Method
          </th>
          <th
            class="SampleControlHtmlTable_HtmlHeaderCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Result
          </th>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getFormFactor()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            1
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            getClient()
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Web
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            userName
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            User Language isRTL
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            undefined
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatCurrency
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            $100,500
          </td>
        </tr>
        <tr
          class="SampleControlHtmlTable_HtmlRow"
        >
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Key"
          >
            formatting formatDateLong
          </td>
          <td
            class="SampleControlHtmlTable_HtmlCell SampleControlHtmlTable_HtmlCell_Value"
          >
            Mon, September 19, 2022
          </td>
        </tr>
      </table>
      <button
        class="SampleControlHtmlTable_ButtonClass"
      >
        Click to invoke: setFullScreen(false)
      </button>
    </div>
  </div>
</body>
`;
